src/lib/snack-bar/snack-bar.ts

```diff
@@ -236,21 +236,27 @@ export class MatSnackBar {
       config.horizontalPosition === 'left' ||
       (config.horizontalPosition === 'start' && !isRtl) ||
       (config.horizontalPosition === 'end' && isRtl));
-    const isRight = !isLeft && config.horizontalPosition !== 'center';
+    const isRight = (!isLeft &&
+      config.horizontalPosition !== 'center' &&
+      config.horizontalPosition !== 'stretch');
     if (isLeft) {
       positionStrategy.left('0');
     } else if (isRight) {
       positionStrategy.right('0');
     } else {
       positionStrategy.centerHorizontally();
     }
-    // Set horizontal position.
+    // Set vertical position.
     if (config.verticalPosition === 'top') {
       positionStrategy.top('0');
     } else {
       positionStrategy.bottom('0');
     }
 
+    if (config.horizontalPosition === 'stretch') {
+      overlayConfig.panelClass = 'mat-snack-bar-stretch';
```

Comment:
In general we try to avoid having multiple APIs for achieving the same thing. In this case you can get the same result by adding the `mat-snack-bar-stretch` class yourself through `MatSnackBar.panelClass`.

Reply:
This is not the same as using `MatSnackBar.panelClass`, I am adding a class to the parent element: the cdk-overlay. I couldn't find a way to add a class to the overlay container besides overriding the MatSnackBar class.

Reply:
Just double-checked and it seems like it's an inconsistency between the dialog, the bottom sheet and the snack bar. For the dialog, adding a `panelClass` means adding a class to the `cdk-overlay-pane`, whereas `panelClass` for the bottom sheet and snack bar means adding a class on the container which is one level down from the `cdk-overlay-pane`.